### DISCUSSION TITLE
Custom scopes return value

### DISCUSSION THREAD

**Discussion by pafernandez-oesia** [laravel/framework#41754] [Q&A] [answered]
Hi!

When using a custom `scope` to retrieve results I've notice that method works if it returns something that is not a `null` value. 

```
public function scopeCustomFilter($query, $value)
{
   return $query->where('column', $value)->first();
}
```

So if you use it in an assignment, and the scope returns `null` you get the prior query `Builder` object assigned:

```
// If customFilter returns values $var would get those values; in other case, $var = $query.
$var = $query->customFilter($value);
```

The question is, is it discouraged to return results from a scope? Should a scope always return a valid query object?

Thank you.

**Comment by Kamandlou** [comment, marked as answer]
It is better not to use a scope for this purpose.

You can do this:‌

```
public function customFilter($value)
{
   return $this->where('column', $value)->first();
}
```

```
$result = ModelName::customFilter($value);
// Or
$result = ModelName::where('column','value')->customFilter($value);
```

**Reply by pafernandez-oesia** [reply]
So scopes should only be used if the method returns a query builder. Thank you.
